mbed_greentea/mbed_target_info.py

```diff
@@ -223,39 +221,50 @@ def get_mbed_targets_from_yotta_local_module(mbed_classic_name, yotta_targets_pa
-            try:
-                with open(path, 'r') as data_file:
-                    target_json_data = json.load(data_file)
-                    yotta_target_name = parse_mbed_target_from_target_json(mbed_classic_name, target_json_data)
-                    if yotta_target_name:
-                        target_dir_name = os.path.join(yotta_targets_path, target_dir)
-                        gt_logger.gt_log_tab("inside '%s' found compatible target '%s'"% (gt_logger.gt_bright(target_dir_name), gt_logger.gt_bright(yotta_target_name)))
-                        result.append(yotta_target_name)
-            except IOError as e:
-                gt_logger.gt_log_err(str(e))
+    if not os.path.exists(yotta_targets_path):
+        return result
+
+    # All local directories with yotta targets
+    target_dirs = [target_dir_name for target_dir_name in os.listdir(yotta_targets_path) if os.path.isdir(os.path.join(yotta_targets_path, target_dir_name))]
+
+    gt_logger.gt_log("local yotta target search in '%s' for compatible mbed-target '%s'"% (gt_logger.gt_bright(yotta_targets_path), gt_logger.gt_bright(mbed_classic_name.lower().strip())))
+
+    for target_dir in target_dirs:
+        path = os.path.join(yotta_targets_path, target_dir, 'target.json')
+        try:
+            with open(path, 'r') as data_file:
+                target_json_data = json.load(data_file)
+                yotta_target_name = parse_mbed_target_from_target_json(mbed_classic_name, target_json_data)
+                if yotta_target_name:
+                    target_dir_name = os.path.join(yotta_targets_path, target_dir)
+                    gt_logger.gt_log_tab("inside '%s' found compatible target '%s'"% (gt_logger.gt_bright(target_dir_name), gt_logger.gt_bright(yotta_target_name)))
+                    result.append(yotta_target_name)
+        except IOError as e:
+            gt_logger.gt_log_err(str(e))
     return result
 
 def parse_mbed_target_from_target_json(mbed_classic_name, target_json_data):
-    result = None
-    if target_json_data and 'keywords' in target_json_data:
-        for keyword in target_json_data['keywords']:
-            # Keywords with mbed-target data embedded in to identify target
-            if keyword.startswith('mbed-target:'):
-                mbed_target, mbed_name = keyword.split(':')
-                if mbed_name.lower() == mbed_classic_name.lower():
-                    # Check in case name of the target is not specified
-                    if 'name' in target_json_data:
-                        yotta_target_name = target_json_data['name']
-                        return yotta_target_name
-    return result
+    if not target_json_data:
+        return None
+
+    if not 'keywords' in target_json_data:
+        return None
+
+    for keyword in target_json_data['keywords']:
+        if not keyword.startswith('mbed-target:'):
+            continue
+
+        mbed_target, mbed_name = keyword.split(':')
```

Comment:
I think this may throw an exception if `keyword.split(':')` does not have 2 elements. Should we be skipping these incorrectly formatted entries?

Reply:
> I think this may throw an exception if keyword.split(':') does not have 2 elements.

It will only throw if the keyword string doesn't contain a colon. The previous line ensures that it will contain one.

> Should we be skipping these incorrectly formatted entries?

Maybe? They aren't necessarily incorrectly formatted, it's just checking that the mbed-target value exists and that it matches the input argument.

Reply:
>     It will only throw if the keyword string doesn't contain a colon. The previous line ensures that it will contain one.

Nope:
```python
>>> a,b = "::".split(":")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack (expected 2)
```

> They aren't necessarily incorrectly formatted,

This implies that split is the wrong thing to use here. How about:
```python
target, _, name = keyword.partition(':')
```
That can't raise an exception, ever:
```python
>>> "".partition(":")
('', '', '')
>>> ":".partition(":")
('', ':', '')
>>> "::::::".partition(":")
('', ':', ':::::')
````